mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -178,15 +180,16 @@ def evaluate_codeblock(self, node: mparser.CodeBlockNode, start: int = 0, end: T
             try:
                 self.current_lineno = cur.lineno
                 self.evaluate_statement(cur)
-            except Exception as e:
-                e.source = self.source # type: ignore
+            except MesonException as e:
+                e.source = self.source
                 if getattr(e, 'lineno', None) is None:
```

Comment:
Since this is now a `MesonException`, we shouldn't need the `getattr` anymore, right?